python/TestHarness/testers/Tester.py

```diff
@@ -312,7 +312,7 @@ def runCommand(self, cmd, cwd, timer, options):
         self.errfile.flush()
 
         # store the contents of output, and close the file
-        self.joined_out = util.readOutput(self.outfile, self.errfile, options)
+        self.joined_out = util.readOutput(self.outfile, self.errfile, options, max_size=self.specs['max_screen_output_size'])
```

Comment:
There are other testers which call readOutput. But I am not sure they need this ability. Perhaps they do (RunException for example).